pypeit/core/procimg.py

```diff
@@ -265,59 +266,104 @@ def gain_frame(amp_img, gain):
     return gain_img
 
 
-def rn_frame(datasec_img, gain, ronoise):
-    """ Generate a RN image
+def rn2_frame(datasec_img, gain, ronoise, units='e-', digitization=True):
+    r"""
+    Construct a readnoise variance image.
+
+    Provided detector readnoise and gain for each detector amplifier, this
+    calculates the combination of the readnoise and digitization (or
+    quantization) noise expected for a single detector readout.  Digitization
+    noise is a fixed :math:`\sqrt{1/12}` ADU [1]_ [2]_, which is typically much
+    smaller than the readnoise, unless the gain is very large.  And, depending
+    on how it was measured, the digitization noise may be incorporated in the
+    documented readnoise of the given instrument.  To exclude the digitization
+    noise from the calculation, set ``digitization=False``.
+
+    The variance calculation in electrons is :math:`V = {\rm RN}^2 +
+    \gamma^2/12`, where RN is the readnoise and :math:`\gamma` is the gain in
+    e-/ADU.  If the requested units are ADU, the returned variance is
+    :math:`V/\gamma^2`.
+    
+    .. [1] `Newberry (1991, PASP, 103, 122) <https://ui.adsabs.harvard.edu/abs/1991PASP..103..122N/abstract>`_
+    .. [2] `Merline & Howell (1995, ExA, 6, 163) <https://ui.adsabs.harvard.edu/abs/1995ExA.....6..163M/abstract>`_
 
-    Parameters
-    ----------
-    datasec_img : ndarray
-        Index image (2D array) assigning each pixel in the data section to an amplifier.
-        A value of 0 means not a data section
-    gain : ndarray, list
-        A list of the gains for each amplifier
-    ronoise : ndarray, list
-        A list of read noise values for each amplifier. If any element of the array is 0.0,
-        the read noise will be determined from the overscan region
+    Args:
+        datasec_img (`numpy.ndarray`_):
+            An integer array indicating the 1-indexed amplifier used to read
+            each pixel in the main data section of the detector.  Values of 0
+            are ignored.  Amplifier numbers are expected sequential and match
+            the number of readnoise and gain values provided.  The shape of this
+            image dictates the shape of the output readnoise variance image.
+        gain (:obj:`float`, array-like):
+            The value of the gain for each amplifier in e-/ADU.  If
+            ``digitization`` is False, this is ignored.
+        ronoise (:obj:`float`, array-like):
+            The value of the readnoise for each amplifier in electrons (e-).  If
+            there is only one amplifier, this can be provided as a single float.
+        units (:obj:`str`, optional):
```

Comment:
I am not sure I see a need to compute this variance in ADU. Generally I see not point of every reducing data in ADU. In my opinion, the moments we read in a file and process any image it should be converted into electrons. But perhaps there is a usage case that I'm missing.

Reply:
I think you mentioned a case over zoom.  So I've left this as an option, but the conversion to counts is now done immediately, as we discussed.